From 63164ca6cae75e08188b9b8bf2872f7368f75228 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 29 Apr 2016 08:45:25 +0200
Subject: [PATCH 4/5] vga: update vga register setup on vbe changes

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1461919526-28456-5-git-send-email-kraxel@redhat.com>
Patchwork-id: 70311
O-Subject: [virt-devel] [RHEL-7.2.z qemu-kvm-rhev PATCH 4/5] vga: update vga register setup on vbe changes
Bugzilla: 1331414
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

Call the new vbe_update_vgaregs() function on vbe configuration
changes, to make sure vga registers are up-to-date.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/display/vga.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/display/vga.c b/hw/display/vga.c
index d7c3399..5a45e55 100644
--- a/hw/display/vga.c
+++ b/hw/display/vga.c
@@ -760,6 +760,7 @@ void vbe_ioport_write_data(void *opaque, uint32_t addr, uint32_t val)
         case VBE_DISPI_INDEX_Y_OFFSET:
             s->vbe_regs[s->vbe_index] = val;
             vbe_fixup_regs(s);
+            vbe_update_vgaregs(s);
             break;
         case VBE_DISPI_INDEX_BANK:
             val &= s->vbe_bank_mask;
-- 
1.8.3.1

